packages/react-components/src/helpers/format-date.js

```diff
@@ -0,0 +1,47 @@
+export const isDateAfter = (date1, date2) => {
+  if (!date1 || !(date1 instanceof Date) || !date2 || !(date2 instanceof Date))
+    return;
+
+  return date1.getTime() > date2.getTime();
+};
+
+export const isDateBefore = (date1, date2) => {
+  if (!date1 || !(date1 instanceof Date) || !date2 || !(date2 instanceof Date))
+    return;
+
+  return date1.getTime() < date2.getTime();
+};
+
+export const isDateSameDay = (date1, date2) => {
+  if (!date1 || !(date1 instanceof Date) || !date2 || !(date2 instanceof Date))
+    return;
+
+  return (
+    date1.getFullYear() === date2.getFullYear() &&
+    date1.getMonth() === date2.getMonth() &&
+    date1.getDay() === date2.getDay()
+  );
+};
+
+export const formatDate = (date, pattern, timezone) => {
+  if (!date || !(date instanceof Date)) return;
+
+  const timeZone = timezone || 'America/New_York';
+
+  // Example: Tuesday, June 9, 2020
+  if (pattern === 'dateFull') {
+    return date.toLocaleString('en-US', { dateStyle: 'full', timeZone });
+  }
+
+  // Example: 10:00 AM
+  if (pattern === 'timeShort') {
+    return date.toLocaleString('en-US', { timeStyle: 'short', timeZone });
+  }
+
+  // Example: Tuesday, June 9, 2020 at 10:00 AM
+  return date.toLocaleString('en-US', {
+    dateStyle: 'full',
+    timeStyle: 'short',
+    timeZone,
+  });
```

Comment:
Having a custom string control the conditional logic for which of the 3 return values we get seems like it might lead to mistakes in the future. What do you thinking about changing the exports so we have something like:

- `formatTime`
- `formatDate`
- `formatDateShort`

And we could have one be the "default", with the other two using it internally:

```js

/**
 * Accepts an optional `options` object matching `Intl.DateTimeFormat` standards.
 * Returns a string with date & time by default.
 * Ex: Tuesday, June 9, 2020 at 10:00 AM
 */
export const formatDate = (
  date,
  options = { dateStyle: 'full', timeStyle: 'short'},
  timeZone = 'America/New_York') => {
    date.toLocaleString('en-US', {...options, timeZone})
}

/**
 * Date only, no time.
 * Ex: Tuesday, June 9, 2020
 */
export const formatDateShort = (date, timezone) => formatDate(date, {dateStyle: 'full'}, timezone)

/**
 * Time only, no date.
 * Ex: 10:00 AM
 */
export const formatTime = (date, timezone) => formatDate(date, {timeStyle: 'short'}, timezone)
```

I haven't tested this, it's just pure speculation right now :sweat_smile:. I'm not totally sold on it either, I'm just seeing what comes to mind.